tests/ServerTest.php

```diff
@@ -223,6 +223,157 @@ public function testParserErrorEmitted()
         $this->connection->expects($this->never())->method('write');
     }
 
+    public function testBodyDataWillBeSendViaRequestEvent()
+    {
+        $server = new Server($this->socket);
+
+        $buffer = '';
+        $server->on('request', function (Request $request, Response $response) use (&$buffer) {
+            $request->on('data', function ($data) use (&$buffer) {
+                $buffer .= $data;
+            });
+        });
+
+        $this->socket->emit('connection', array($this->connection));
+
+        $data = "GET / HTTP/1.1\r\n";
+        $data .= "Host: example.com:80\r\n";
+        $data .= "Connection: close\r\n";
+        $data .= "Content-Length: 5\r\n";
+        $data .= "\r\n";
+        $data .= "hello";
+
+        $this->connection->emit('data', array($data));
+
+        $this->assertEquals('hello', $buffer);
+    }
+
+    public function testChunkedEncodedRequestWillBeParsedForRequestEvent()
+    {
+        $server = new Server($this->socket);
+
+        $buffer =  '';
+        $server->on('request', function (Request $request, Response $response) use (&$buffer) {
+            $request->on('data', function ($data) use (&$buffer) {
+                $buffer .= $data;
+            });
+        });
+
+        $this->socket->emit('connection', array($this->connection));
+
+        $data = "GET / HTTP/1.1\r\n";
+        $data .= "Host: example.com:80\r\n";
+        $data .= "Connection: close\r\n";
+        $data .= "Transfer-Encoding: chunked\r\n";
+        $data .= "\r\n";
+        $data .= "5\r\nhello\r\n";
+        $data .= "0\r\n\r\n";
+
+        $this->connection->emit('data', array($data));
+
+        $this->assertEquals('hello', $buffer);
+    }
+
+    public function testChunkedEncodedRequestAdditionalDataWontBeEmitted()
+    {
+        $server = new Server($this->socket);
+
+        $buffer =  '';
+        $server->on('request', function (Request $request, Response $response) use (&$buffer) {
+            $request->on('data', function ($data) use (&$buffer) {
+                $buffer .= $data;
+            });
+        });
+
+        $this->socket->emit('connection', array($this->connection));
+
+        $data = "GET / HTTP/1.1\r\n";
+        $data .= "Host: example.com:80\r\n";
+        $data .= "Connection: close\r\n";
+        $data .= "Transfer-Encoding: chunked\r\n";
+        $data .= "\r\n";
+        $data .= "5\r\nhello\r\n";
+        $data .= "0\r\n\r\n";
+        $data .= "2\r\nhi\r\n";
+
+        $this->connection->emit('data', array($data));
+        $this->assertEquals('hello', $buffer);
```

Comment:
Can probably use `expectCallableOnceWith('hello');` here? Also, can you add tests for the `end`, `error` and `close` events as well? (See also other test cases)